app/controllers/api/v1/resources_controller.rb

```diff
@@ -0,0 +1,85 @@
+module Api
+  module V1
+    class ResourcesController < ApplicationController
+      before_action :authenticate_user!, except: [:index, :show]
+      before_filter :set_resource, only: [:show, :update, :destroy]
+
+      def index
+        resources = Resource.with_tags params[:tags]
+
+        render json: resources, status: :ok
+      rescue StandardError => e
+        render json: { errors: e.message }, status: :unprocessable_entity
+      end
+
+      def show
+        render json: @resource, status: :ok
+      rescue StandardError => e
+        render json: { errors: e.message }, status: :unprocessable_entity
+      end
+
+      def create
+        ActiveRecord::Base.transaction do
+          resource = Resource.new resource_params
+          resource.tag_list.add(params[:tags], parse: true) if params[:tags].present?
```

Comment:
You have a method `permitted_params` below. Is it possible to call that instead of params directly?